Blog: How to measure product-market fit (#6414)
PostHog/posthog.com: contents/blog/how-to-measure-product-market-fit.md

---
date: 2023-06-08
title: "In-depth: How to reliably measure product-market fit"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author: ["andy-vandervell"]
featuredImage: ../images/blog/pmf-game-guide/pmf-guide.png
featuredImageType: full
category: Startups
tags:
  - Guides
---

[Achieving product-market fit](/blog/product-market-fit-game) isn't a golden ticket to a billion-dollar IPO, but startups can't succeed without it. 

Most guides tell you how it _feels_ to have product-market fit. Everything feels effortless when you have it, or impossibly difficult when you don't, and so on.

But product-market fit isn't just an ephemeral gut feeling. You can _measure_ it, and _it moves_ as your customer's needs change, or new solutions appear.

This guide is about finding ways to measure product-market fit, your progress toward it, and how it changes over time.

> ## Leading and lagging indicators

Reviewer: I really enjoyed reading the sections on each of individual metrics, but I felt that this section dragged on a bit. It reads a bit bland, like a generic blog that's trying to explain to me what product metrics are. Can we shorten it so we get to the good stuff sooner?

Perhaps cutting out the parts where it mentions the differences from Leading and Lagging indicators – it's not clear to me why I would need to know the difference?

Measuring product-market fit requires a _combination_ of leading and lagging indicators.

**Leading indicators**, are predictive. A surge in organic growth and positive word-of-mouth suggests product-market fit, but can't confirm it. They include:
- **Organic word-of-mouth growth** – Not directly linked to paid promotions
- **Meaningful user engagement** – Users experiencing real value    
- **User feedback & PMF Survey** – Direct feedback from users

**Lagging indicators** confirm whether you have product-market fit. They're more reliable, particularly if they're linked to financial performance. They include:
- **Retention curve** – Evidence users are coming back
- **Burn multiple** – Evidence you're growing sustainably
- **CLV/CAC ratio** – Can you acquire customers for less than they cost to support? 

> We recommend a minimum of three to start with – likely two leading indicators and one lagging indicator – and adding more when you need.

Reviewer: Why? How do I choose which ones?

Author: This is answered in the next paragraph, tbf!

**Pre-revenue**, word-of-mouth, user engagement, and retention are good indicators of product-market fit. They'll tell you how people feel about your product, whether they're using it, and if they stick around.

> **Post-revenue** you'll want to want another lagging indicator, like burn multiple or CLV/CAC ratio, that provide a more concrete link to revenue and the financial viability of your product.

Author: Note to tweak these a bit so it's not just pre/post revenue.

Why a minimum of three? Because of what Brian Balfour, a serial co-founder and a former VP of Growth at Hubspot, calls The Trifecta:[^1]

![pmf trifecta](../images/blog/how-to-measure-product-market-fit/pmf-trifecta.png)

If you can nail these three at the same time, it's a strong sign you have product-market fit.

> ## Indicator #1: Organic word-of-mouth growth

Reviewer: Really enjoyed this section on the different ways of measuring word of mouth growth! Very actionable

To be clear... you _can't_ validate product-market fit _using word-of-mouth alone_. That way lies madness. But it is a useful leading indicator _when confirmed using other metrics_, such as user engagement, and retention.

Hopefully, you already have a strong grasp of your organic user growth via user signup, or whatever metric makes sense for your product.

But you can, with a little work, also track word of mouth awareness and sentiment of your product. Here are a few options.

### 1. Brand mention alerts

Use a tool like [Syften](https://syften.com/) to monitor social media, communities, newsletters, etc., for mentions of any keyword you choose, such as your brand name. We use Syften at PostHog to send alerts to a `#brand-alerts` channel on our Slack. 

You can go one step further by setting up a Zapier integration with Google Sheets, where you can collect your mentions, and track trends over time.

Brand alerts are also a great way to drive acquisition by talking to people considering your product. 

> > **❗️Important:** It's _much easier_ to track mentions if your company has a unique name. You're bang out of luck if it's a common noun – e.g. Apple, Amplitude. 

Reviewer: ![image](https://github.com/PostHog/posthog.com/assets/7090054/950cd6cd-92c0-4a89-b00c-b49088e77e8b)

It's also useful for comparing yourself to other companies, such as those you _know_ have product-market fit, or competitors.

![google trend comparison](../images/blog/how-to-measure-product-market-fit/posthog-with-logrocket.png)

At PostHog, we also use [Glimpse](https://meetglimpse.com/) – a handy extension that augments Trends by adding trend lines, forecasts, and real search volumes. 

![trends with Glimpse](../images/blog/how-to-measure-product-market-fit/glimpse.png)

It'll also suggest similar companies you can compare yourself against.

![related companies](../images/blog/how-to-measure-product-market-fit/related-companies.png)

**Google Search Console** is an alternative if you don't have a unique brand name, but the data isn't 100% reliable, so you're better off using the next method.

### 3. Organic traffic to your homepage

One final way to track word-of-mouth is organic users who visit your website homepage as a proxy for word-of-mouth. We track this in PostHog as an insight that combines:

- Unique users who visited the homepage via Google search
- Unique users how visited the homepage directly

> This captures people who search for your brand, and those who navigate there directly, or visit via Slack, WhatsApp, etc. 

Author: ```suggestion
This captures people who visit our homepage by searching for us, typing in our website address, or via "dark social" sources like Slack and WhatsApp, while excluding existing users who have an identified email address.
```

> This method will capture some existing users as well, but we've found it correlates well with surges in word-of-mouth awareness.

Author: ```suggestion
This method will capture some existing users as well, but we've found it correlates well with surges in word-of-mouth awareness.

![posthog proxy](../images/blog/how-to-measure-product-market-fit/word-of-mouth-proxy.png)

Above you can see the trend for our homepage since the beginning of 2021. Unsurprisingly, it's similar to that of  Google Trends. 
```

> 💡 **PostHog Tip:** If you're using PostHog, remember to add `UTM Source` = `Is not set` to filter out any paid ad campaigns you're running.

### When to use word-of-mouth

> Word-of-mouth growth is a good option for any product-led company – i.e. a product that's self-serve and doesn't do outbound sales. It's a weak leading indicator, but useful so long as you don't buy into your own PR and verify with other metrics.

Author: ```suggestion
Word-of-mouth growth is a good option for any product-led company – i.e. a product that's self-serve and doesn't do outbound sales. It's a weak leading indicator, but useful so long as you don't buy into your own PR and verify it with other metrics, such as user signups or engagement.
```

| <span className="text-green text-lg">✔</span> Easy to track                                                              | <span className="text-red text-lg">✖</span> Not a reliable sign of product-market fit on its own                         |
| <span className="text-green text-lg">✔</span> Useful when you're pre-revenue                                             | <span className="text-red text-lg">✖</span> Spikes in awareness and user growth can be misleading                        |
| <span className="text-green text-lg">✔</span> Easy to compare with other companies                                       | <span className="text-red text-lg">✖</span> Not useful for companies with low-profile brands – e.g. enterprise products  |
| <span className="text-green text-lg">✔</span> Positive feedback and growth is highly motivating                          | <span className="text-red text-lg">✖</span> It's tempting (and easy) to rationalize negative word-of-mouth               |

> ## Indicator #2: User feedback & PMF Survey

Author: ```suggestion
## Indicator #2: User feedback & PMF Survey

- **Type:** Leading
- ** Good for:** Most B2B products
```

- a) Very disappointed
- b) Somewhat disappointed
- c) Not disappointed

You want to learn everything you possibly can about people who answer "very disappointed". Organize a call and talk to these users. Ask open-ended, probing questions to understand why they love your product. Do the same for users in the "somewhat disappointed" cohort.

Based on his research of 100+ startups, Ellis believes 40% answering "very disappointed" is a strong signal of product-market fit. The more responses you get, the more reliable the signal. Ellis recommends a minimum of 30:

> "In my experience, a minimum of 30 responses is needed before the survey becomes directionally useful. At 100+ responses I am much more confident in the results." – Sean Ellis[^2]

An open research project run by Hiten Shah[^3], co-founder of KISSmetrics, used the PMF Survey on 731 Slack users, the results showed:

- 51% responded they would be very disappointed if they couldn't use Slack
- These users believed it increased productivity and improved collaboration
- All groups said video conferencing was a must-have addition

It's no coincidence Slack has since added video conferencing.

> ### A PMF Survey case study

Reviewer: We should link to the superhuman post here. It's probably the most famous blog on the topic, and people would find it interesting to read

| <span className="text-green text-lg">✔</span> Can help guide product development in the right direction                  | <span className="text-red text-lg">✖</span> PMF Survey is somewhat labor-intensive                                       |
| <span className="text-green text-lg">✔</span> Easy to break down into multiple cohorts to understand your ICP            | <span className="text-red text-lg">✖</span> Needs to be backed-up with 121 interviews                                    |
| <span className="text-green text-lg">✔</span> Can be systematized to improve product-market fit over time                | <span className="text-red text-lg">✖</span> Not driven by real usage – always validate with engagement or retention data |
| <span className="text-green text-lg">✔</span> Useful for tracking how your product-market fit changes over time          | <span className="text-red text-lg">✖</span> Requires a minimum number of responses to be directionally useful            |
| <span className="text-green text-lg">✔</span> Will help you understand _why_ you do or don't have product-market fit     |               |

> ## Indicator #3: Meaningful user engagement

Author: ```suggestion
## Indicator #3: Meaningful user engagement

**Type:** Leading indicator
**Good for:** Any software product
```

| <span className="text-green text-lg">✔</span> Tracks the real value users gain from your product                         | <span className="text-red text-lg">✖</span> High engagement isn't necessarily predictive of revenue                      |
| <span className="text-green text-lg">✔</span> Passive to maintain – set it up once and track forever                     | <span className="text-red text-lg">✖</span> Can provide a false positive if you're tracking the wrong things             |
| <span className="text-green text-lg">✔</span> Easy to understand for all teams                                           | <span className="text-red text-lg">✖</span> Requires user feedback or session replays to understand behavior             |
| <span className="text-green text-lg">✔</span> Can be broken down by user cohorts for comparison                          |           |
| <span className="text-green text-lg">✔</span> You can A/B test ways to improve user engagement                           |           |

> ## Indicator #4: Retention curve

Author: ```suggestion
## Indicator #4: Retention curve

**Type:** Lagging indicator
**Good for:** Any software product
```

|--------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------| 
| <span className="text-green text-lg">✔</span> Reliable lagging indicator of product-market fit                           | <span className="text-red text-lg">✖</span> Not predictive of revenue – will users actually pay for your product?        |
| <span className="text-green text-lg">✔</span> Can be broken down by cohort for comparison                                | <span className="text-red text-lg">✖</span> You need several weeks or months of data to get a reliable signal            |
| <span className="text-green text-lg">✔</span> Easy to understand – flat is good, not flat is bad!                        | <span className="text-red text-lg">✖</span> Won't help you understand why you do or don't have product-market fit        |

> ## Indicator #5: The burn multiple

Author: ```suggestion
## Indicator #5: The burn multiple

- **Type:** Lagging indicator
- **Good for:** Late-stage companies seeking investment
```

| <span className="text-green text-lg">✔</span> It's easy to calculate – it's just two numbers any founder should have     | <span className="text-red text-lg">✖</span> It's not useful if you're pre-revenue                                        |
| <span className="text-green text-lg">✔</span> A steady decline is a strong indicator of product-market fit               | <span className="text-red text-lg">✖</span> It won't help you understand why you do or don't have product-market fit     |
| <span className="text-green text-lg">✔</span> It's useful for investors                                                  | <span className="text-red text-lg">✖</span> Skewed by non-product factors – e.g. poor cost control                       |
| <span className="text-green text-lg">✔</span> It's easy to understand                                                    |        |
| <span className="text-green text-lg">✔</span> Useful for mid to late-stage startups                                      |        |

> ## Indicator #6 CLV/CAC ratio

Author: ```suggestion
## Indicator #6 CLV/CAC ratio

- **Type:** Lagging indicator
- **Good for:** Late-stage sales-led companies
```

---
date: 2023-06-08
title: "In-depth: How to reliably measure product-market fit"
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
author: ["andy-vandervell"]
featuredImage: ../images/blog/pmf-game-guide/pmf-guide.png
featuredImageType: full
category: Startups
tags:
  - Guides
---

[Achieving product-market fit](/blog/product-market-fit-game) isn't a golden ticket to a billion-dollar IPO, but startups can't succeed without it. 

> Most guides tell you how it _feels_ to have product-market fit. Everything feels effortless when you have it, or impossibly difficult when you don't, and so on.

Reviewer 2: Many people talk about "the pull" of product market fit, but what does that actually look like metrics wise?

https://www.lennysnewsletter.com/i/3111833/sign-sudden-and-significant-pull-from-the-market

> Most guides tell you how it _feels_ to have product-market fit. Everything feels effortless when you have it, or impossibly difficult when you don't, and so on.

Author: RE: 1, I think it does. At least, it's a fairly commonly held feeling.

On the second, that's kind of what the word-of-mouth section is about, but maybe need to highlight the "pull feeling" a bit more as a sign of PMF there.

But product-market fit isn't just an ephemeral gut feeling. You can _measure_ it, and _it moves_ as your customer's needs change, or new solutions appear.

This guide is about finding ways to measure product-market fit, your progress toward it, and how it changes over time.

## Leading and lagging indicators

Measuring product-market fit requires a _combination_ of leading and lagging indicators.

**Leading indicators**, are predictive. A surge in organic growth and positive word-of-mouth suggests product-market fit, but can't confirm it. They include:
- **Organic word-of-mouth growth** – Not directly linked to paid promotions
- **Meaningful user engagement** – Users experiencing real value    
- **User feedback & PMF Survey** – Direct feedback from users

**Lagging indicators** confirm whether you have product-market fit. They're more reliable, particularly if they're linked to financial performance. They include:
- **Retention curve** – Evidence users are coming back
- **Burn multiple** – Evidence you're growing sustainably

> - **CLV/CAC ratio** – Can you acquire customers for less than they cost to support? 

Author: Yeah, I went back and forth on this. People seem to use it interchangeably. Will switch this.

**Post-revenue** you'll want to want another lagging indicator, like burn multiple or CLV/CAC ratio, that provide a more concrete link to revenue and the financial viability of your product.

Why a minimum of three? Because of what Brian Balfour, a serial co-founder and a former VP of Growth at Hubspot calls The Trifecta:[^1]

1. Non-trivial growth
2. Meaningful usage
3. User retention

[TURN ABOVE INTO GRAPHIC]

If you can nail these three at the same time, it's a strong sign you have product-market fit.

## Indicator #1: Organic word-of-mouth growth

To be clear... you _can't_ validate product-market fit _using word-of-mouth alone_. That way lies madness. But it is a useful leading indicator _when confirmed using other metrics_, such as user engagement, and retention.

Hopefully, you already have a strong grasp of your organic user growth via user signup, or whatever metric makes sense for your product.

But you can, with a little work, also track word of mouth awareness and sentiment of your product. Here are a few options.

> ### 1. Brand mention alerts

Author: Can add this. For brand mentions, it's more about sentiment than a curve.

Also, mental note to me add something around what "good growth" in signups is for PMF.

| <span className="text-green text-lg">✔</span> Easy to compare with other companies                                       | <span className="text-red text-lg">✖</span> Not useful for companies with low-profile brands – e.g. enterprise products  |
| <span className="text-green text-lg">✔</span> Positive feedback and growth is highly motivating                          | <span className="text-red text-lg">✖</span> It's tempting (and easy) to rationalize negative word-of-mouth               |

## Indicator #2: User feedback & PMF Survey

Like word-of-mouth, _consistently_ positive user feedback is an indicator of product-market fit. There's a decent chance you're onto a winner when people tell you they can't live without your product.

To this end, the PMF Survey is a twist on the classic Net Promoter Score, but it's designed _specifically_ for finding product-market fit, and creating a feedback loop for user feedback.

Created by entrepreneur Sean Ellis, the core question (among others) it asks is:

“How would you feel if you could no longer use [ProductName]?”

- a) Very disappointed
- b) Somewhat disappointed
- c) Not disappointed

> You want to learn everything you possibly can about people who answer "very disappointed". Organize a call and talk to these users. Ask open-ended, probing questions to understand why they love your product. Do the same for users in the "somewhat disappointed" cohort.

Reviewer 3: Classic followup article - '802 unintuitive signs you have product-market fit'

|--------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------| 
| <span className="text-green text-lg">✔</span> Reliable lagging indicator of product-market fit                           | <span className="text-red text-lg">✖</span> Not predictive of revenue – will users actually pay for your product?        |
| <span className="text-green text-lg">✔</span> Can be broken down by cohort for comparison                                | <span className="text-red text-lg">✖</span> You need several weeks or months of data to get a reliable signal            |
| <span className="text-green text-lg">✔</span> Easy to understand – flat is good, not flat is bad!                        | <span className="text-red text-lg">✖</span> Won't help you understand why you do or don't have product-market fit        |

## Indicator #5: The burn multiple

> Created by former PayPal COO David Sacks, burn multiple measures how much you're spending to grow revenue. The less you spend, the more efficient your growth. More efficient growth = stronger product-market fit.

Reviewer 2: I think you need a little more on why burn is important to product-market fit before going into calculating it.

> Created by former PayPal COO David Sacks, burn multiple measures how much you're spending to grow revenue. The less you spend, the more efficient your growth. More efficient growth = stronger product-market fit.

Reviewer 3: Anecdata - at PostHog we don't care about burn multiple (yet). We see it as something to care about in the context of efficiency/unit economics, not PMF.

> Created by former PayPal COO David Sacks, burn multiple measures how much you're spending to grow revenue. The less you spend, the more efficient your growth. More efficient growth = stronger product-market fit.

Author: So yeah, I'm glad you both picked up on this as this is where the article becomes more "debatable". I think the point in including burn multiple and the next one is they confirm how _strong_ your PMF is, particularly to outside investors – i.e. we may not care about it, but people looking from the outside might. This is basically Sacks' point.

Will have another look at this section and see I can communicate this better.

| <span className="text-green text-lg">✔</span> It's easy to understand                                                    |        |
| <span className="text-green text-lg">✔</span> Useful for mid to late-stage startups                                      |        |

## Indicator #6 CLV/CAC ratio

> **CLV** = customer lifetime value – i.e. the total value of a customer after churn and the cost of supporting / retaining them.
>
> **CAC** = customer acquisition cost - i.e. total sales and marketing spend divided by how many customers you acquired.

CLV/CAC ratio is a robust but complex measure of product-market fit. Robust because it speaks directly to the long-term financial health of your company; complex because calculating CLV accurately isn't a trivial task.

Customer lifetime value attempts to quantify how much a customer is worth over their entire lifetime. This means you need to understand:

- Your average deal size.
- How long customers stay until they churn.
- Your gross margin – i.e. how much it costs to support and retain customers.
- How long it'll take for a customer to pay back the cost of acquiring them. 

> Consequently, the ratio is best used by companies selling to enterprises, where sales costs are higher and play a larger role in a product's success.  

Reviewer 2: How can early stage (pre-product market fit) companies figure out CLV?

Reviewer 2: Again this section needs a bit more on why this matters to product-market fit, seems like a later stage metric.

Author: It definitely is a later stage metric, I probably need to make that clearer. This will go in the conclusion, but the TL;DR is validate PMF with "The Trifecta" first, then add the ratio or burn multiple as a test.

| <span className="text-green text-lg">✔</span> Reliable lagging indicator of product-market fit                           | <span className="text-red text-lg">✖</span> Not predictive of revenue – will users actually pay for your product?        |
| <span className="text-green text-lg">✔</span> Can be broken down by cohort for comparison                                | <span className="text-red text-lg">✖</span> You need several weeks or months of data to get a reliable signal            |
| <span className="text-green text-lg">✔</span> Easy to understand – flat is good, not flat is bad!                        | <span className="text-red text-lg">✖</span> Won't help you understand why you do or don't have product-market fit        |

## Indicator #5: The burn multiple

- **Type:** Lagging indicator
- **Good for:** Late-stage companies seeking investment

Created by former PayPal COO David Sacks, burn multiple measures how much you're spending to grow revenue. The less you spend, the more efficient your growth. More efficient growth = stronger product-market fit.

> Because it measures efficiency rather than product-market fit directly, it's best used as a way to prove how _strong_ your product-market fit is. For this reason, it's especially useful for later-stage startups seeking further investment.

Author: ```suggestion
Because it measures efficiency rather than product-market fit directly, it's best used as a way to prove how _strong_ your product-market fit is. It's especially useful for later-stage startups seeking further investment, but it's a board-level metric – i.e. product teams won't find it useful.
```
